integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/common/BenchUtils.scala

```diff
@@ -150,34 +150,46 @@ object BenchUtils {
     val queryStartTime = Instant.now()
 
     val queryPlansWithMetrics = new ListBuffer[SparkPlanNode]()
+    val exceptions = new ListBuffer[String]()
 
     var df: DataFrame = null
     val queryTimes = new ListBuffer[Long]()
     for (i <- 0 until iterations) {
 
-      // capture spark plan metrics on the final run
-      if (i+1 == iterations) {
-        spark.listenerManager.register(new BenchmarkListener(queryPlansWithMetrics))
+      // capture spark plan metrics on the first run
+      if (i == 0) {
+        spark.listenerManager.register(new BenchmarkListener(queryPlansWithMetrics, exceptions))
       }
 
       println(s"*** Start iteration $i:")
       val start = System.nanoTime()
-      df = createDataFrame(spark)
-
-      resultsAction match {
-        case Collect() => df.collect()
-        case WriteCsv(path, mode, options) =>
-          df.write.mode(mode).options(options).csv(path)
-        case WriteOrc(path, mode, options) =>
-          df.write.mode(mode).options(options).orc(path)
-        case WriteParquet(path, mode, options) =>
-          df.write.mode(mode).options(options).parquet(path)
-      }
+      try {
+        df = createDataFrame(spark)
+
+        resultsAction match {
+          case Collect() => df.collect()
+          case WriteCsv(path, mode, options) =>
+            df.write.mode(mode).options(options).csv(path)
+          case WriteOrc(path, mode, options) =>
+            df.write.mode(mode).options(options).orc(path)
+          case WriteParquet(path, mode, options) =>
+            df.write.mode(mode).options(options).parquet(path)
+        }
 
-      val end = System.nanoTime()
-      val elapsed = NANOSECONDS.toMillis(end - start)
-      queryTimes.append(elapsed)
-      println(s"*** Iteration $i took $elapsed msec.")
+        val end = System.nanoTime()
+        val elapsed = NANOSECONDS.toMillis(end - start)
+        queryTimes.append(elapsed)
+        println(s"*** Iteration $i took $elapsed msec.")
+
+      } catch {
+        case e: Exception =>
+          val end = System.nanoTime()
+          val elapsed = NANOSECONDS.toMillis(end - start)
+          println(s"*** Iteration $i failed after $elapsed msec.")
+          queryTimes.append(-1)
+          exceptions.append(e.toString)
+          e.printStackTrace()
```

Comment:
I think `e.toString` may omit the stack trace itself, even though it's printed. Do you think it makes sense to include them in the report? In that case we can print to a stream or writer using: https://docs.oracle.com/javase/7/docs/api/java/lang/Throwable.html#printStackTrace(java.io.PrintStream).

Reply:
So far I was able to get away with this approach because the exceptions are generally coming from the executors and the stack trace is already part of the message, but you're right, it would be better to do the full stack trace. I'll make that change today.